### DISCUSSION TITLE
Syslog plugin - facility selection

### DISCUSSION THREAD

**Discussion by jideel** [Kong/kong#7315] [Ideas and feature requests]
### Select the facility used by the syslog plugin

It would be great if it were possible to configure the syslog Facility used by the plugin.
It's currently hardcoded to lsyslog.FACILITY_USER : 
_/usr/local/share/lua/5.1/kong/plugins/syslog , line 31_ 
`l_open(SENDER_NAME, lsyslog.FACILITY_USER)`

Maybe a setting like "config.facility", similar to those described here : https://docs.konghq.com/hub/kong-inc/syslog/ ?

We are using the facility to "route" messages to a centralized log server, and would like to avoid to manually patch the plugin every deployment.

Thank you,
Best regards

**Comment by bungle** [comment]
Thanks for suggestion. Would you mind sending a PR for the addition?

**Comment by jideel** [comment]
OK, i already have working code and patch files against schema.lua and handler.lua.
I'm using 2.0.4, so i'm not sure between which versions the PR should be made : 2.0.4 vs master ?
Thanks

**Comment by bungle** [comment]
@jideel, It is a new feature, so open it against `next`. Thank you!
